Add 5.11 release notes (#384)
chef/cookstyle: RELEASE_NOTES.md

The `PoiseArchiveUsage` cop detects the use of the `poise_archive` resource in cookbooks. Chef Infra Client 15+ ships with the built-in `archive_file` resource which should be used instead.

`Enabled by default`: True

`Autocorrects`: No

#### ChefModernize/Definitions

The `Definitions` cop detects cookbooks that include Definitions. We highly recommend replacing legacy Definitions with Custom Resource. Definitions are not *currently* deprecated, but we do plan to deprecate this functionality in the future. See [Converting Definitions to Custom Resources](https://docs.chef.io/definitions.html) for more information on the benefits of Custom Resources and how to convert legacy Definitions.

`Enabled by default`: True

`Autocorrects`: No

#### ChefEffortless/SearchForEnvironmentsOrRoles

The `SearchForEnvironmentsOrRoles` cop is an optional cop for users migrated to Policyfiles or the Effortless pattern that detects the use of search find nodes based in their roles and environments. Policyfiles replaces roles and environments and these node searches will need to be updated as part of a migration.

`Enabled by default`: False

`Autocorrects`: No

### Other fixes and changes

> - The RuboCop engine that powers Cookstyle has been upgraded from 0.72 to 0.75.1. This update includes 50+ bug fixes that may result in new warnings during you Cookstyle runs. We have also enabled a new `Migration/DepartmentName` cop which will update any rubocop disable comments in code to use updated cop names. This prevents cop name deprecation warnings when you run Cookstyle.

Reviewer: ```suggestion
- The RuboCop engine that powers Cookstyle has been upgraded from 0.72 to 0.75.1. This update includes more than 50 bug fixes that may result in new warnings during your Cookstyle runs. We have also enabled a new `Migration/DepartmentName` cop that will detect and update the names of disabled cops in cookbook code. This will prevent cop name deprecation warnings when you run Cookstyle.
```
